File: CSE 240/hw14/question1.cpp
#include <iostream>

using namespace std;

int foo(int x, int y) // size-n problem
{
	if (x <= 0) // stopping condition
		return y;
	else if (y <= 0) // stopping condition
		return x;
	if (x >= y)
		return x + foo(x - 1, y - 2); // size-m problem and soltion of size-n problem
	else
		return y + foo(x - 2, y - 3); // size-m problem and soltion of size-n problem
}

void main()
{
	cout << foo(5, 6) << endl;



}

File: CSE 240/hw14/question1.pl
foo(X,Y,Z):- % stopping condition
	Y =< 0,
	Z is X.

foo(X,Y,Z):- % stopping condition 
	X =< 0,
	Z is Y.

foo(X,Y,Z):- % size-n problem
	X >= Y,
	X1 is X - 1,
	Y1 is Y - 2,
	foo(X1, Y1, Z1), % size-m problem 
	Z is X + Z1. % soltion of size-n problem

foo(X,Y,Z):- % size-n problem
	X < Y,
	X1 is X -2,
	Y1 is Y -3,
	foo(X1, Y1, Z1), % size-m problem 
	Z is Y + Z1. % soltion of size-n problem
 	

